app/collections/collaborators.jsx

```diff
@@ -0,0 +1,375 @@
+import React from 'react';
+import apiClient from 'panoptes-client/lib/api-client';
+import UserSearch from '../components/user-search';
+import checkIfCollectionOwner from '../lib/check-if-collection-owner';
+import alert from '../lib/alert';
+
+const ID_PREFIX = 'COLLECTION_COLLABORATORS_PAGE_';
+
+const POSSIBLE_ROLES = [
+  'owner',
+  'collaborator',
+  'contributor',
+  'viewer'
+];
+
+const ROLES_INFO = {
+  owner: {
+    label: 'Owner',
+    description: 'The collection creator. There can only be one.'
+  },
+  collaborator: {
+    label: 'Collaborator',
+    description: 'Collaborators have full access to add and remove subjects from the collection.'
+  },
+  contributor: {
+    label: 'Contributor',
+    description: 'Contributors can only add subjects to the collection.'
+  },
+  viewer: {
+    label: 'Viewer',
+    description: 'Viewers can see this collection even if it\'s private.'
+  }
+};
+
+export class RoleCreator extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      creating: null,
+      error: null
+    };
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+
+    const checkboxes = this.node.querySelectorAll('[name="role"]');
+    const users = this.userSearch.value().map(option => parseInt(option.value, 10));
+    const checkedBoxes = Object.keys(checkboxes).filter(key => checkboxes[key].checked);
+    const roles = checkedBoxes.map(value => checkboxes[value].value);
+
+    this.setState({
+      creating: true,
+      error: null
+    });
+
+    const roleSets = users.map(id =>
+      apiClient.type('collection_roles').create({
+        roles,
+        links: {
+          collection: this.props.collection.id,
+          user: id
+        }
+      })
+    );
+
+    Promise.all((roleSets.map((roleSet) => {
+      return roleSet.save().then(role => role);
+    })))
+      .then((args) => {
+        this.userSearch.clear();
+        Object.keys(checkboxes).map((key) => {
+          checkboxes[key].checked = false;
+        });
+        if (this.props.onAdd) {
+          this.props.onAdd.apply(null, args);
+        }
+      })
+      .catch((error) => {
+        this.setState({ error });
+      })
+      .then(() => {
+        this.setState({ creating: false });
+      });
+  }
+
+  render() {
+    let style;
+    let errorMessage;
+
+    if (this.state.creating) {
+      style = {
+        opacity: 0.5,
+        pointerEvents: 'none'
+      };
+    }
+
+    if (this.state.error) {
+      const rawErrorMessage = this.state.error.toString();
+
+      if (rawErrorMessage.indexOf('No User with id')) {
```

Comment:
This was written well before we fixed the client to just pass the error object rather than returning a string. I would presume that a user not found would 404, so you can check specifically for that, `if (error.status === 404)` instead of looking for a string of text.